test/scaling.jl

```diff
@@ -274,32 +279,33 @@
 
             @testset "dims = 1" begin
                 scaling = MeanStdScaling(M; dims=1)
-                @test FeatureTransforms.apply(M, scaling; dims=1) ≈ [0.0 -0.707107 -0.707107; 0.0 0.707107 0.707107] atol=1e-5
+                @test FeatureTransforms.apply(M, scaling; dims=1) ≈ M_expected atol=1e-5
             end
 
             @testset "dims = 2" begin
                 scaling = MeanStdScaling(M; dims=2)
-                @test FeatureTransforms.apply(M, scaling; dims=2) ≈ [0.0 -1.0 1.0; -1.0 0.0 1.0] atol=1e-5
+                @test FeatureTransforms.apply(M, scaling; dims=2) ≈ M_expected atol=1e-5
             end
 
             @testset "inds" begin
                 scaling = MeanStdScaling(M)
                 @test FeatureTransforms.apply(M, scaling; inds=[2, 3]) ≈ [-0.559017, -1.11803] atol=1e-5
                 @test FeatureTransforms.apply(M, scaling; dims=:, inds=[2, 3]) ≈ [-0.559017, -1.11803] atol=1e-5
 
-                scaling = MeanStdScaling(M; dims=1)
-                @test FeatureTransforms.apply(M, scaling; dims=1, inds=[2]) ≈ [0.0 0.707107 0.707107] atol=1e-5
+                scaling = MeanStdScaling(M; dims=1, inds=[2])
```

Comment:
Interesting that an unwrapped index now works:

```julia
julia> scaling = MeanStdScaling(M; dims=1, inds=[2])
MeanStdScaling(1.0, 1.0)

julia> scaling = MeanStdScaling(M; dims=1, inds=2)
MeanStdScaling(1.0, 1.0)

julia> FeatureTransforms.apply(M, scaling; dims=1, inds=[2])
1×3 Array{Float64,2}:
-1.0  0.0  1.0

julia> FeatureTransforms.apply(M, scaling; dims=1, inds=2)
3-element Array{Float64,1}:
 -1.0
  0.0
  1.0
```

The closest equivalent before (though the refactor and #39 make it not really comparable):

```julia
julia> scaling = MeanStdScaling(M; dims=2)
MeanStdScaling((1 = 0.0, 2 = 1.0), (1 = 0.5, 2 = 1.0))

julia> FeatureTransforms.apply(M, scaling; dims=2, inds=[2])
2×1 Array{Float64,2}:
 -1.0
  0.0

julia> FeatureTransforms.apply(M, scaling; dims=2, inds=2)
ERROR: MethodError: no method matching _apply(::Float64, ::MeanStdScaling; name=Symbol("1"))
Closest candidates are:
  _apply(::Any, ::Transform, ::Union{Tuple, AbstractArray}; kwargs...) at /Users/bencottier/JuliaEnvs/Transform.jl/src/transformers.jl:121
  _apply(::Any, ::Transform, ::Any; kwargs...) at /Users/bencottier/JuliaEnvs/Transform.jl/src/transformers.jl:117
  _apply(::Any, ::OneHotEncoding{R}; kwargs...) where R<:Real at /Users/bencottier/JuliaEnvs/Transform.jl/src/one_hot_encoding.jl:35
  ...
Stacktrace:
 [1] (::FeatureTransforms.var"#3#4"{Int64,Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},MeanStdScaling})(::Array{Float64,1}) at /Users/bencottier/JuliaEnvs/Transform.jl/src/transformers.jl:82
 [2] mapslices(::FeatureTransforms.var"#3#4"{Int64,Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},MeanStdScaling}, ::Array{Float64,2}; dims::Int64) at ./abstractarray.jl:2083
 [3] #apply#2 at /Users/bencottier/JuliaEnvs/Transform.jl/src/transformers.jl:80 [inlined]
 [4] top-level scope at REPL[5]:1
```